pkg/cmd/admin/policy/reconcile_clusterroles.go

```diff
@@ -88,49 +83,61 @@ func NewCmdReconcileClusterRoles(name, fullName string, f *clientcmd.Factory, ou
 	return cmd
 }
 
-func (o *reconcileClusterOptions) Complete(cmd *cobra.Command, f *clientcmd.Factory, args []string) error {
-	if len(args) != 0 {
-		return errors.New("No arguments are allowed")
-	}
-
+func (o *reconcileClusterOptions) Complete(cmd *cobra.Command, f *clientcmd.Factory) error {
 	oclient, _, err := f.Clients()
 	if err != nil {
 		return err
 	}
 	o.RoleClient = oclient.ClusterRoles()
 
+	o.Output = kcmdutil.GetFlagString(cmd, "output")
+
 	return nil
 }
 
-func (o *reconcileClusterOptions) ReplaceChangedRoles(changedRoles []*authorizationapi.ClusterRole) error {
-	for i := range changedRoles {
-		role, err := o.RoleClient.Get(changedRoles[i].Name)
-		if err != nil && !kapierrors.IsNotFound(err) {
-			return err
-		}
+func (o *reconcileClusterOptions) Validate(args []string) error {
+	if len(args) != 0 {
+		return errors.New("no arguments are allowed")
+	}
+	if o.RoleClient == nil {
+		return errors.New("a role client is required")
+	}
+	if o.Output != "yaml" && o.Output != "json" && o.Output != "" {
+		return fmt.Errorf("unknown output specified: %s", o.Output)
+	}
+	return nil
+}
 
-		if kapierrors.IsNotFound(err) {
-			createdRole, err := o.RoleClient.Create(changedRoles[i])
-			if err != nil {
-				return err
-			}
+// RunReconcileClusterRoles contains all the necessary functionality for the OpenShift cli reconcile-cluster-roles command
+func (o *reconcileClusterOptions) RunReconcileClusterRoles(cmd *cobra.Command, f *clientcmd.Factory) error {
+	changedClusterRoles, err := o.ChangedClusterRoles()
+	if err != nil {
+		return err
+	}
 
-			fmt.Fprintf(o.Out, "clusterrole/%s\n", createdRole.Name)
-			continue
+	if len(changedClusterRoles) == 0 {
+		return nil
+	}
+
+	if (len(o.Output) != 0) && !o.Confirmed {
+		list := &kapi.List{}
+		for _, item := range changedClusterRoles {
+			list.Items = append(list.Items, item)
 		}
 
-		role.Rules = changedRoles[i].Rules
-		updatedRole, err := o.RoleClient.Update(role)
-		if err != nil {
+		if err := f.Factory.PrintObject(cmd, list, o.Out); err != nil {
 			return err
 		}
+	}
 
-		fmt.Fprintf(o.Out, "clusterrole/%s\n", updatedRole.Name)
+	if o.Confirmed {
+		return o.ReplaceChangedRoles(changedClusterRoles)
 	}
 
 	return nil
 }
 
+// ChangedClusterRoles returns all the roles that diverge from the recommended bootstrap policy
```

Comment:
This godoc isn't quite right.  This returning the role that must be created or replaced to satisfy the recommended bootstrap policy.  The mutation of rules means that the role itself won't necessarily match anything.